### PR TITLE
[RELEASE] 0.1.0

### PR DESCRIPTION
**Pull request by adrw**
* Paired with @swankjesse to prepare Barber for first release: `0.1.0`.
* Renamed subproject from `client` to `barber`
* Added a `mkdocs` docs site that includes `dokka` KDocs similar to https://square.github.io/okhttp/
* Updated `releasing.md` with more thorough instructions on preparation work required to publish to Sonatype

### CODE REVIEW

File: barber/src/main/kotlin/app/cash/barber/BarbershopBuilder.kt [hunk 1]
```diff
@@ -176,7 +176,8 @@ class BarbershopBuilder : Barbershop.Builder {
     // Pre-compile Mustache templates
     val documentDataFields: MutableMap<String, Mustache?> =
       documentTemplate.fields.mapValues {
-        mustacheFactory.compile(StringReader(it.value), it.value)
+        val mustache = mustacheFactory.compile(StringReader(it.value), it.value)
+        mustache
```
Comment (wesleyk) [inline barber/src/main/kotlin/app/cash/barber/BarbershopBuilder.kt]: why this change?

Comment (adrw) [reply barber/src/main/kotlin/app/cash/barber/BarbershopBuilder.kt]: Thanks, will revert. Had set the variable for debugging during brief API review.

File: build.gradle [hunk 2]
```diff
@@ -79,6 +83,17 @@ subprojects {
...
     testRuntimeOnly dep.junitEngine
   }
+
+  // We have to set the dokka configuration after evaluation since the com.vanniktech.maven.publish
+  // plugin overwrites our dokka configuration on projects where it's applied.
+  afterEvaluate { p ->
+    p.tasks.dokka {
+      reportUndocumented = false
+      skipDeprecated = true
+      jdkVersion = 8
```
Comment (wesleyk) [inline build.gradle]: why not 11?

Comment (adrw) [reply build.gradle]: @swankjesse : 11 doesn't give us much at this point over 8 with respect to the docs

Comment (swankjesse) [reply build.gradle]: 11 is more strict about some things and I’m not sure whether it works or not. Worth trying if there’s anything from 11 we need.

File: docs/css/app.css [hunk 3]
```diff
@@ -0,0 +1,48 @@
+@font-face {
```
Comment (swankjesse) [inline docs/css/app.css]: nice
